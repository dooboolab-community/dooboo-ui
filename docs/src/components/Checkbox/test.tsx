import {Button, Checkbox, Hr, ThemeProvider} from 'dooboo-ui';
import type {FC, ReactElement} from 'react';
import {FontAwesome, createIconSetFromIcoMoon} from '@expo/vector-icons';
import {IC_FACEBOOK, IC_GOOGLE} from '../../icon';
import {Image, View} from 'react-native';
import React, {useState} from 'react';
import styled, {css} from '@emotion/native';

import {action} from '@storybook/addon-actions';
import {useFonts} from 'expo-font';

const collectingFontIconSelection = require('./selection.json');

const Icon = createIconSetFromIcoMoon(
  collectingFontIconSelection,
  'IcoMoon',
  './doobooui.ttf',
);

const StoryContainer = styled.View`
  flex: 1;
  align-self: stretch;
`;

const StyledText = styled.Text`
  margin: 8px;
`;

const ScrollContainer = styled.ScrollView`
  width: 100%;
`;

const CheckboxDefault: FC = () => {
  const [fontsLoaded] = useFonts({
    IcoMoon: require('./doobooui.ttf'),
  });

  // if (!fontsLoaded) return <View />;

  return (
    <StoryContainer>
      <ScrollContainer
        contentContainerStyle={{
          paddingVertical: 60,
          paddingHorizontal: 40,
          justifyContent: 'center',
          alignItems: 'center',
        }}>
        <Checkbox checked={true} type="danger" />
      </ScrollContainer>
    </StoryContainer>
  );
};

const Default = (): ReactElement => (
  <ThemeProvider>
    <>
      <Icon name="like-solid" color="black" size={32} />
      <FontAwesome name="plus" />
      <CheckboxDefault />
    </>
  </ThemeProvider>
);

export default Default;
