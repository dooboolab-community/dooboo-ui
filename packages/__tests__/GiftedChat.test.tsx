import 'react-native';

import * as React from 'react';

import type {RenderAPI} from '@testing-library/react-native';
import {render} from '@testing-library/react-native';

import GiftedChatInput from '../GiftedChat';
import {createTestProps} from '../../test/testUtils';

let props: any;
let component: React.ReactElement;
let testingLib: RenderAPI;

jest.useFakeTimers();

describe('[GiftedChatInput] render', () => {
  beforeEach(() => {
    props = createTestProps({
      messages: [
        {
          id: '',
          sender: {
            id: '0',
            nickname: 'sender111',
            thumbURL: '',
            photoURL: '',
            statusMessage: '',
          },
          message: 'hello1',
        },
        {
          id: '',
          sender: {
            id: '2',
            nickname: 'sender111',
            thumbURL: '',
            photoURL: '',
            statusMessage: '',
          },
          message:
            'Hello2. This is long message. This is long message.This is long message.' +
            'This is long message. This is long message. This is long message.' +
            'This is long message. This is long message.' +
            'This is long message. This is long message. This is long message.',
        },
        {
          id: '',
          sender: {
            id: '0',
            nickname: 'sender111',
            thumbURL: '',
            photoURL: '',
            statusMessage: '',
          },
          message: 'hello',
        },
      ],
    });

    component = <GiftedChatInput {...props} />;
  });

  it('renders without crashing', () => {
    testingLib = render(component);

    const json = testingLib.toJSON();

    expect(json).toMatchSnapshot();
  });

  // describe('interactions', () => {
  //   beforeEach(() => {
  //     testingLib = render(component);
  //   });

  //   it('should call [setShowMenu] when focused', () => {
  //     const textInput = testingLib.getByTestId('input-chat');

  //     textInput.props.onFocus();
  //   });

  //   it('should [showMenu] when touch pressed', () => {
  //     let touchMenu = testingLib.getByTestId('touch-menu');

  //     fireEvent.press(touchMenu);

  //     touchMenu = testingLib.getByTestId('touch-menu');
  //     fireEvent.press(touchMenu);
  //   });

  //   it('should call [setShowMenu] when focused again', () => {
  //     const touchMenu = testingLib.getByTestId('touch-menu');

  //     fireEvent.press(touchMenu);
  //   });

  //   it('should invoke changeText event handler when message changed', () => {
  //     const textInput = testingLib.getByTestId('input-chat');

  //     jest.runAllTimers();
  //     fireEvent.changeText(textInput, 'chat test');
  //     // TODO: expect should pass
  //     // expect(textInput.props.value).toEqual('chat test');
  //   });
  // });
});
