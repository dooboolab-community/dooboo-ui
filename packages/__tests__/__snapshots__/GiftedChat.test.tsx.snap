// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[GiftedChatInput] render renders without crashing 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Array [
        Object {
          "alignItems": "center",
          "alignSelf": "stretch",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
        },
        undefined,
      ],
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <RCTScrollView
    ListEmptyComponent={[Function]}
    ListHeaderComponent={
      <View
        style={
          Object {
            "height": 28,
          }
        }
      />
    }
    contentContainerStyle={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    invertStickyHeaders={true}
    inverted={true}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "scaleY": -1,
            },
          ],
        },
        Object {
          "alignSelf": "stretch",
        },
      ]
    }
    viewabilityConfigCallbackPairs={Array []}
  >
    <View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "transform": Array [
              Object {
                "scaleY": -1,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "height": 28,
            }
          }
        />
      </View>
      <View />
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderTopWidth": 0.3,
          "flexBasis": 0,
          "flexDirection": "row",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "space-between",
          "maxHeight": 56,
          "maxWidth": "100%",
          "minHeight": 56,
          "paddingLeft": 8,
          "paddingRight": 8,
          "width": "100%",
        },
        Object {
          "backgroundColor": undefined,
          "borderColor": undefined,
        },
      ]
    }
  >
    <TextInput
      defaultValue=""
      multiline={true}
      onChangeText={[Function]}
      onFocus={[Function]}
      placeholder=""
      style={
        Array [
          Object {
            "color": "black",
            "fontSize": 14,
            "paddingBottom": 4,
            "paddingLeft": 48,
          },
          Object {
            "backgroundColor": undefined,
            "color": undefined,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginRight": 10,
            "paddingBottom": 5,
            "paddingTop": 5,
          },
        ]
      }
      testID="input-chat"
      value=""
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": "100%",
          "justifyContent": "center",
          "left": 10,
          "minWidth": 20,
          "opacity": 1,
          "position": "absolute",
        }
      }
      testID="touch-menu"
    >
      <View />
    </View>
    <View
      style={
        Object {
          "flexGrow": 0,
          "flexShrink": 0,
          "marginVertical": 8,
        }
      }
    >
      <View />
    </View>
  </View>
</View>
`;
